WPFHospitalEditor/AdditionalInformation.xaml.cs

```diff
@@ -88,20 +90,41 @@ private void ModifyDynamicWPFGrid(string[] contentRows)
 
         private void createRowContent(string[] contentRows)
         {
-            for (int i = 0; i < contentRows.Length; i++)
+            if (PatientIsLogged())
             {
-                if (contentRows[i].Equals(""))
-                    break;
-
-                string[] label = contentRows[i].Split("=");
-                labelContent.Add(label[0]);
-                value.Add(label[1]);
+                if(StringContainsWorkingHours(contentRows[0])) setRowContent(0);
+            } else
+            {
+                for (int i = 0; i < contentRows.Length; i++)
+                {
+                    if (contentRows[i].Equals("")) break;
+                    setRowContent(i);
+                }
             }
             insertData();
             addCancelButton(contentRows);
             addOkButton(contentRows);
         }
 
+        private bool PatientIsLogged()
+        {
+            if (role.Equals("patient")) return true;
```

Comment:
you use the role in many other places. You should definitely consider using enumeration. The problem with using a string is that in case you want to modify a string for a particular role, you would have to do it manually in many different places.